vector/src/main/java/im/vector/widgets/WidgetsManager.java

```diff
@@ -489,40 +489,42 @@ public void onLiveEvent(MXSession session, Event event) {
      * @param callback the callback
      */
     public static void getFormattedWidgetUrl(final Context context, final Widget widget, final ApiCallback<String> callback) {
-        getScalarToken(context, Matrix.getInstance(context).getSession(widget.getSessionId()), new ApiCallback<String>() {
-            @Override
-            public void onSuccess(String token) {
-                if (null == token) {
-                    callback.onSuccess(widget.getUrl());
-                } else {
-                    callback.onSuccess(widget.getUrl() + "&scalar_token=" + token);
+        if (isScalarUrl(context, widget.getUrl())) {
+            getScalarToken(context, Matrix.getInstance(context).getSession(widget.getSessionId()), new SimpleApiCallback<String>(callback) {
+                @Override
+                public void onSuccess(String token) {
+                    if (null == token) {
+                        callback.onSuccess(widget.getUrl());
+                    } else {
+                        callback.onSuccess(UrlExtensionsKt.appendParamToUrl(new StringBuilder(widget.getUrl()), "scalar_token", token).toString());
+                    }
                 }
-            }
+            });
+        } else {
+            // Do not provide the scalar token
+            callback.onSuccess(widget.getUrl());
+        }
+    }
 
-            @Override
-            public void onNetworkError(Exception e) {
-                if (null != callback) {
-                    callback.onNetworkError(e);
-                }
-            }
+    /**
+     * Return true if the url is allowed to receive the scalar token in parameter
+     *
+     * @param context
+     * @param url
+     * @return true if the url is allowed to receive the scalar token in parameter
+     */
+    public static boolean isScalarUrl(Context context, String url) {
+        String[] array = context.getResources().getStringArray(R.array.integrations_widgets_urls);
```

Comment:
riot-web (and riot-ios) fallbacks to `integrations_rest_url` if `integrations_widgets_urls` is empty

Reply:
ok, done